test/install_script/nvm_check_global_modules

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+cleanup () {
+  rm -rf "$npm_config_prefix/lib" >/dev/null 2>&1
+  unset npm_config_prefix
+
+  rm -f npm
+  PATH="$ORIGINAL_PATH"
+
+  unset -f setup cleanup die skip
+  unset message ORIGINAL_PATH
+}
+die () { echo $@ ; cleanup ; exit 1; }
+
+NVM_ENV=testing . ../../install.sh
+
+setup () {
+  ORIGINAL_PATH="$PATH"
+
+  npm_config_prefix="$(pwd)"
+  export npm_config_prefix
+  mkdir -p "$npm_config_prefix/lib"
+}
+
+
+setup
+
+npm install -g nop >/dev/null
+message=$(nvm_check_global_modules)
+[ ! -z "$message" ] || die "nvm_check_global_modules should have printed a notice when npm had global modules installed"
```

Comment:
1. Was intentionally using the same check, and negating it, amongst the various tests; I thought that was clearer to a non-initiate (I tend to believe the tests should be accessible to people that the code being tested isn't; as tests should be the entry-point for learning about the codebase). Let me know if this is unwelcome practice, I'll push a switch for `-z` and `-n` instead of `-z` and `! -z`!

2. I just pushed a bunch of debugging-related output, including showing the actual output of the check command on test failure. Covers this amongst other things~